superset/common/query_object.py

```diff
@@ -0,0 +1,59 @@
+# pylint: disable=R
+from typing import Dict, List, Optional, Union
+
+from superset import app
+from superset.utils import core as utils
+
+# TODO: Type Metrics dictionary with TypedDict when it becomes a vanilla python type
+# https://github.com/python/mypy/issues/5288
+Metric = Union[str, Dict]
+
+
+class QueryObject:
+    """
+    The query object's schema matches the interfaces of DB connectors like sqla
+    and druid. The query objects are constructed on the client.
+    """
+    def __init__(
+            self,
+            granularity: str,
+            groupby: List[str],
+            metrics: List[Metric],
+            filters: List[str],
+            time_range: Optional[str] = None,
+            since: Optional[str] = None,
+            until: Optional[str] = None,
+            time_shift: Optional[str] = None,
+            is_timeseries: bool = False,
```

Comment:
do we need the `is_timeseries` flag?

Reply:
Yeah, I think so. We're currently setting it to different values depending on which viz it is on the server side. My thought is we'll compute this on the client and pass it in as part of the `query_object`. Thoughts?

Reply:
yes, we do need it, and it is one field of the query_object.

Reply:
@conglei long-term is the plan to remove it?